server/src/main/java/org/elasticsearch/index/shard/RemoveCorruptedShardDataCommand.java

```diff
@@ -0,0 +1,584 @@
+                        terminal.println("");
+                        try {
+                            indexCleanStatus = removeCorruptedLuceneSegmentsAction.getCleanStatus(shardPath, indexDir,
+                                writeIndexLock, printStream, verbose);
+                        } catch (Exception e) {
+                            terminal.println(e.getMessage());
+                            throw e;
+                        }
+
+                        terminal.println("");
+                        terminal.println(" >> Lucene index is " + indexCleanStatus.v1().getMessage() + " at " + indexPath);
+                        terminal.println("");
+                    } else {
+                        indexCleanStatus = Tuple.tuple(CleanStatus.CLEAN, null);
+                    }
+
+                    ////////// Translog
+                    // as translog relies on data stored in an index commit - we have to have non unrecoverable index to truncate translog
+                    if (indexCleanStatus.v1() != CleanStatus.UNRECOVERABLE) {
+                        terminal.println("");
+                        terminal.println("Opening translog at " + translogPath);
+                        terminal.println("");
+                        try {
+                            translogCleanStatus = truncateTranslogAction.getCleanStatus(shardPath, indexDir);
+                        } catch (Exception e) {
+                            terminal.println(e.getMessage());
+                            throw e;
+                        }
+
+                        terminal.println("");
+                        terminal.println(" >> Translog is " + translogCleanStatus.v1().getMessage() + " at " + translogPath);
+                        terminal.println("");
+                    } else {
+                        translogCleanStatus = Tuple.tuple(CleanStatus.UNRECOVERABLE, null);
+                    }
+
+                    ////////// Drop corrupted data
+                    final CleanStatus indexStatus = indexCleanStatus.v1();
+                    final CleanStatus translogStatus = translogCleanStatus.v1();
+
+                    if (indexStatus == CleanStatus.CLEAN && translogStatus == CleanStatus.CLEAN) {
+                        throw new ElasticsearchException("Shard does not seem to be corrupted at " + shardPath.getDataPath());
+                    }
+
+                    if (indexStatus == CleanStatus.UNRECOVERABLE) {
+                        if (indexCleanStatus.v2() != null) {
+                            terminal.println("Details: " + indexCleanStatus.v2());
+                        }
+
+                        terminal.println("You can allocate completely empty primary shard with command: ");
+
+                        printRerouteCommand(shardPath, terminal, false);
+
+                        throw new ElasticsearchException("Index is unrecoverable - there are missing segments");
```

Comment:
Is "there are missing segments" the only reason why an index could be unrecoverable? I suggest removing the explanation and just saying "Index is unrecoverable" since we've already printed details above.